### ISSUE TITLE
Search result: address_component[0] with empty types => MapperException

### ISSUE THREAD

**Issue by NilsBendfeldt** [bradcornford/Googlmapper#203]
We stumbled over a few addresses where the location search results in a MapperException, which is apperently caused by an empty _types_ list in the first _address_component_.

maps api result looks like this:

```
   "results" : [
      {
         "address_components" : [
            {
               "long_name" : "Dr.-Paul-Rosenbaum-Platz",
               "short_name" : "Dr.-Paul-Rosenbaum-Platz",
               "types" : []
            },
```

The Google Maps Developer Guide says:
> The types[] array indicates the type of the returned result. This array contains a set of **zero** or more tags identifying the type of feature returned in the  result...

We think its a valid search result and therefor should not cause an exception in Mapper.php:

```
if (!isset($resultObject->results[0]->formatted_address) ||

        //this check fails ?
	!isset($resultObject->results[0]->address_components[0]->types[0]) || 

	!isset($resultObject->results[0]->geometry->location->lat) ||
	!isset($resultObject->results[0]->geometry->location->lng) ||
	!isset($resultObject->results[0]->place_id) ||
	isset($resultObject->status) && $resultObject->status != self::GOOGLE_RESPONSE_OK
) {
    throw new MapperException('The location search return invalid result data.');
}
```

Examples:
https://maps.googleapis.com/maps/api/geocode/json?address=Dr.-Paul-Rosenbaum-Platz%2C+59227+Ahlen&sensor=false&key=

https://maps.googleapis.com/maps/api/geocode/json?address=Georg-Hellmair-Platz%2C+86899+Landsberg+am+Lech&sensor=false&key=

**Comment by bradcornford** [comment]
Hi there, 

Looking at this, you are correct, types always exist, but can contain no items. I'll get a patch put together for this.

**Comment by bradcornford** [comment]
I have now pushed a fix to current versions v2.28.1/v1.23.1.
